src/mlpack/tests/string_encoding_test.cpp

```diff
@@ -0,0 +1,1017 @@
+
+  encoder.Encode(stringEncodingInput, output, tokenizer);
+
+  const DictionaryType& dictionary = encoder.Dictionary();
+
+  // Checking that everything is mapped to different numbers
+  std::unordered_map<size_t, size_t> keysCount;
+  for (auto& keyValue : dictionary.Mapping())
+  {
+    keysCount[keyValue.second]++;
+    // Every token should be mapped only once
+    BOOST_REQUIRE_EQUAL(keysCount[keyValue.second], 1);
+  }
+
+  vector<vector<double>> expected = {
+    {  1.40546510810816, 1.40546510810816, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 1, 2.09861228866811, 1.40546510810816,
+       1.40546510810816, 1.40546510810816, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 1, 2.09861228866811, 2.09861228866811, 0, 0, 0, 0, 0,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       0  },
+    {  0, 1.40546510810816, 0, 0, 0, 2, 0, 0, 4.21639532432449,
+       4.21639532432449, 0, 0, 0, 3, 0, 0, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 6.29583686600433, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 2.09861228866811, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 2.09861228866811, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 2.09861228866811, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0  },
+    {  1.40546510810816, 0, 0, 0, 0, 1, 0, 1.40546510810816, 0, 0, 0, 0, 0, 1,
+       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+       2.09861228866811, 2.09861228866811, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 2.09861228866811, 2.09861228866811, 2.09861228866811,
+       2.09861228866811, 2.09861228866811, 2.09861228866811  }
+  };
+  for (size_t i = 0; i < expected.size(); i++)
+    for (size_t j = 0; j < expected[i].size(); j++)
+      BOOST_REQUIRE_CLOSE(expected[i][j], output[i][j], 1e-12);
+}
+
+/**
+ * Test TFIDF encoding for characters using lamda function, using rawcount as
+ * tf type and smoothidf as false.
+ */
+BOOST_AUTO_TEST_CASE(RawcountTfIdfEncodingIndividualCharactersTest)
+{
+  vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  TfIdfEncoding<CharExtract::TokenType> encoder(
+      (TfIdfEncodingPolicy(0, false)));
+
+  // Passing a empty string to encode charactersrawcountsmoothidftrue
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 1.4054651081081644, 2, 2.8109302162163288, 0, 0 },
+    { 0, 2, 2.8109302162163288, 2.8109302162163288, 2.0986122886681100 },
+    { 1.4054651081081644, 1, 0, 1.4054651081081644, 0 }
+  };
+  CheckMatrices(output, target, 1e-12);
+}
+
+/**
+ * Test the one pass modification of the Tf Idf encoding algorithm
+ * in case of individual character encoding, using raw count as type,
+ * and smoothidf as false.
+ */
+BOOST_AUTO_TEST_CASE(OnePassRawcountEncodingIndividualCharactersTest)
```

Comment:
It's not the one pass modification. It's exactly the same algorithm. But the output is a vector.